R/internals.R

```diff
@@ -0,0 +1,121 @@
+#' Find the path to the root folder within a factory
+#'
+#' This function is used to find the fully expanded root path within a factory.
+#'
+#' @param directory the full path to the directory you are interested in.  If
+#'   NULL (default) the function assumes you are already in a factory and will
+#'   return the address of the containing factory root.
+#'
+#' @noRd
+factory_root <- function(directory = NULL) {
```

Comment:
Same comment as elsewhere (can't remember where, sorry); do we want to default to 
`directory = "."` or `directory = getwd()` here and remove the leading `if` statement?

Reply:
closed with https://github.com/reconhub/reportfactory/pull/68/commits/e60198f05faaf899a33aab96f3d9634ab3413252